runtime/src/main/java/org/corfudb/runtime/view/TableRegistry.java

```diff
@@ -878,4 +887,198 @@ public void shutdown() {
+        TableDescriptor<K, V, M> build(Class<K> kClass, Class<V> vClass, Class<M> mClass, TableOptions tableOpts) {
+            return TableDescriptor
+                    .<K, V, M>builder()
+                    .kClass(kClass)
+                    .vClass(vClass)
+                    .mClass(mClass)
+                    .withSchema(mClass != null)
+                    .tableOptions(tableOpts)
+                    .build();
+        }
+
+        public static <K extends Message, V extends Message, M extends Message>
+        TableDescriptor<K, V, M> build(Class<K> kClass, Class<V> vClass, Class<M> mClass) {
+            TableOptions defaultOpts;
+            try {
+                defaultOpts = TableOptions.fromProtoSchema(vClass);
+            } catch (Exception e) {
+                throw new IllegalStateException(e);
+            }
+
+            return build(kClass, vClass, mClass, defaultOpts);
+        }
+
+        public SchemaOptions getSchemaOptions() {
+            return tableOptions.getSchemaOptions();
+        }
+
+        public V getDefaultValueMessage() {
+            return getDefaultMessage(vClass);
+        }
+
+        public K getDefaultKeyMessage() {
+            return getDefaultMessage(kClass);
+        }
+
+        private static <T> T getDefaultMessage(Class<T> clazz) {
+            try {
+                return ClassUtils.cast(clazz.getMethod(DEFAULT_INSTANCE_METHOD_NAME).invoke(null));
+            } catch (Exception ex) {
+                throw new IllegalStateException(DEFAULT_METHOD_NOT_FOUND_ERR_MSG);
+            }
+        }
+
+        public void metadataHandler(Consumer<M> defaultMetadataHandler) {
+            if (mClass == null) {
+                return;
+            }
+
+            M defaultMessage = getDefaultMetadataMessage();
+            defaultMetadataHandler.accept(defaultMessage);
+        }
+
+        private M getDefaultMetadataMessage() {
+            if (mClass == null) {
+                return null;
+            }
+
+            return getDefaultMessage(mClass);
+        }
+
+        public Object[] getArgs() {
+            if (withSchema) {
+                return new Object[]{getIndexer()};
+            } else {
+                return new Object[]{};
+            }
+        }
+
+        public ProtobufIndexer getIndexer() {
+            SchemaOptions schemaOptions = getSchemaOptions();
+            V msg = getDefaultValueMessage();
+            return new ProtobufIndexer(msg, schemaOptions);
+        }
+    }
+
+    @Builder
+    @EqualsAndHashCode
+    @ToString
+    public static class FullyQualifiedTableName {
+        @Default
+        private final Optional<String> namespace = Optional.empty();
+        private final String tableName;
+
+        /**
+         * Fully qualified table name created to produce the stream uuid.
+         *
+         * @return Fully qualified table name.
+         */
+        public String toFqdn() {
```

Comment:
Fqdn typically stands for fully qualified *domain* name - it has a well defined meaning in networking terminology - I don't think it is a good idea to overload that term to refer to a table name right?

Reply:
fully qualified domain name ...... for a table!

But, I can rename it, any ideas about giving it better name?

Reply:
maybe `toFqtn` ?

Reply:
sounds weird :)